ast/src/groups/group_value.rs

```diff
@@ -24,12 +24,15 @@ use serde::{Deserialize, Serialize};
 use std::fmt;
 use tendril::StrTendril;
 
+/// A group literal.
 #[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
 pub enum GroupValue {
+    /// Single group literal, e.g., `42group`.
```

Comment:
If you take the earlier suggestion, also here single => product, and a few lines below tuple => affine.